File: java/Gate Seven/TestNumberGuessingApp.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;
import java.util.Scanner; 


class TestNumberGuessingApp {

    @Test
    void testRandomNumberGenerator() {

        boolean actual = (NumberGuessingApp.randomNumberGenerator() <= 100 && NumberGuessingApp.randomNumberGenerator() >= 1);
        assertTrue(actual);
  
    }

    @Test
    void testTrackAndDisplayRatingOne() {

        String actual = NumberGuessingApp.trackAndDisplayRating(2);
	String expected = "Excellent";
        assertEquals(actual, expected);
  
    }
    @Test
    void testTrackAndDisplayRatingTwo() {

        String actual = NumberGuessingApp.trackAndDisplayRating(3);
	String expected = "Good";
        assertEquals(actual, expected);
  
    }

    @Test
    void testTrackAndDisplayRatingFail() {

        String actual = NumberGuessingApp.trackAndDisplayRating(6);
	String expected = "Better luck";
        assertEquals(actual, expected);
  
    }

    @Test
    void testInputValidator() {
	Scanner userInput = new Scanner(System.in);
        boolean actual = NumberGuessingApp.inputValidator(userInput);
        assertTrue(actual);
  
    }
}

File: python/Gate Seven/TestNumberGuessingApp.py
import unittest
from number_guessing_app import *

class TestNumberGuessingApp(unittest.TestCase):
    def test_random_number_generator(self):
        actual = (random_number_generator() >= 1 and random_number_generator() <= 100)
        expected = True
        self.assertEqual(actual, expected)
        
    def test_track_and_display_rating_one(self):
        actual = track_and_display_rating(2)
        expected = "Excellent"
        self.assertEqual(actual, expected)

    def test_track_and_display_rating_two(self):
        actual = track_and_display_rating(3)
        expected = "Good"
        self.assertEqual(actual, expected)

    def test_track_and_display_rating_fail(self):
        actual = track_and_display_rating(6)
        expected = "Better luck"
        self.assertEqual(actual, expected)


    def test_number_guessing_game(self):
        actual = number_guessing_game(20)
        expected = 5
        self.assertEqual(actual, expected)
        

    def test_input_validator_with_a_number(self):
        actual = input_validator("5")
        expected = True
        self.assertEqual(actual, expected)

    def test_input_validator_with_a_letter(self):
        actual = input_validator("e")
        expected = False
        self.assertEqual(actual, expected)


    def test_input_validator_out_of_range(self):
        actual = input_validator("200")
        expected = False
        self.assertEqual(actual, expected)

    def test_deisplay_summary(self):
        actual = deisplay_summary(20, 3)
        expected = "The Correct number is 20, you took 3 attempts, Good"
        self.assertEqual(actual, expected)

    def test_deisplay_summary_failure(self):
        actual = deisplay_summary(50, 6)
        expected = "The Correct number is 50, you took 6 attempts, Better luck"
        self.assertEqual(actual, expected)






        
